From b5f583e72aa480c544e13622d86f8a45bb90a528 Mon Sep 17 00:00:00 2001
From: Esteban Gutierrez <esteban@cloudera.com>
Date: Mon, 7 Dec 2015 15:52:14 -0800
Subject: [PATCH 106/217] CLOUDERA-BUILD update dependency listing for cdh5
 profile.

* move some dep scopes back to compile from test
* add cdh5 profile where missing
* ensure compilation against mrv1

Change-Id: I56d7cfa3e9968d9cc1db8b3a5654bf57caa2e0a2
Reason: build
Ref: CDH-23876
---
 hbase-client/pom.xml         |   14 +++++---------
 hbase-hadoop2-compat/pom.xml |   10 ----------
 hbase-it/pom.xml             |    5 +++++
 hbase-prefix-tree/pom.xml    |   17 ++++++++++++++++-
 hbase-rest/pom.xml           |   27 +++++++++++++++++++++++++--
 hbase-server/pom.xml         |   12 ++++++++----
 pom.xml                      |    1 -
 7 files changed, 59 insertions(+), 27 deletions(-)

diff --git a/hbase-client/pom.xml b/hbase-client/pom.xml
index 627feb9..f870da2 100644
--- a/hbase-client/pom.xml
+++ b/hbase-client/pom.xml
@@ -153,6 +153,11 @@
       <artifactId>commons-logging</artifactId>
     </dependency>
     <dependency>
+       <groupId>com.google.code.findbugs</groupId>
+       <artifactId>jsr305</artifactId>
+       <optional>true</optional>
+    </dependency>
+    <dependency>
       <groupId>com.google.guava</groupId>
       <artifactId>guava</artifactId>
     </dependency>
@@ -246,11 +251,6 @@
       </activation>
       <dependencies>
         <dependency>
-           <groupId>com.google.code.findbugs</groupId>
-           <artifactId>jsr305</artifactId>
-           <optional>true</optional>
-        </dependency>
-        <dependency>
           <groupId>org.apache.hadoop</groupId>
           <artifactId>hadoop-auth</artifactId>
         </dependency>
@@ -386,10 +386,6 @@
         </dependency>
         <dependency>
           <groupId>org.apache.hadoop</groupId>
-          <artifactId>hadoop-auth</artifactId>
-        </dependency>
-        <dependency>
-          <groupId>org.apache.hadoop</groupId>
           <artifactId>hadoop-core</artifactId>
           <scope>provided</scope>
           <exclusions>
diff --git a/hbase-hadoop2-compat/pom.xml b/hbase-hadoop2-compat/pom.xml
index 9a023a6..8855de7 100644
--- a/hbase-hadoop2-compat/pom.xml
+++ b/hbase-hadoop2-compat/pom.xml
@@ -172,16 +172,6 @@ limitations under the License.
       <artifactId>commons-math</artifactId>
     </dependency>
     <dependency>
-      <groupId>org.apache.hadoop</groupId>
-      <artifactId>hadoop-mapreduce-client-core</artifactId>
-      <version>${hadoop-two.version}</version>
-    </dependency>
-    <dependency>
-      <groupId>org.apache.hadoop</groupId>
-      <artifactId>hadoop-common</artifactId>
-      <version>${hadoop-two.version}</version>
-    </dependency>
-    <dependency>
       <groupId>com.yammer.metrics</groupId>
       <artifactId>metrics-core</artifactId>
     </dependency>
diff --git a/hbase-it/pom.xml b/hbase-it/pom.xml
index 98da6ad..dc87488 100644
--- a/hbase-it/pom.xml
+++ b/hbase-it/pom.xml
@@ -394,6 +394,11 @@
           <groupId>org.apache.hadoop</groupId>
           <artifactId>hadoop-common</artifactId>
         </dependency>
+        <dependency>
+          <groupId>org.apache.hadoop</groupId>
+          <artifactId>hadoop-test</artifactId>
+          <scope>test</scope>
+        </dependency>
       </dependencies>
       <build>
         <plugins>
diff --git a/hbase-prefix-tree/pom.xml b/hbase-prefix-tree/pom.xml
index 93b0b12..55b915e 100644
--- a/hbase-prefix-tree/pom.xml
+++ b/hbase-prefix-tree/pom.xml
@@ -144,6 +144,21 @@
   </dependencies>
 
   <profiles>
+    <profile>
+      <id>cdh5</id>
+      <activation>
+        <property>
+            <!--Below formatting for dev-support/generate-hadoopX-poms.sh-->
+            <!--h2--><name>!hadoop.profile</name>
+        </property>
+      </activation>
+      <dependencies>
+        <dependency>
+          <groupId>org.apache.hadoop</groupId>
+          <artifactId>hadoop-common</artifactId>
+        </dependency>
+      </dependencies>
+    </profile>
     <!-- Profiles for building against different hadoop versions -->
     <profile>
       <id>hadoop-1.1</id>
@@ -184,7 +199,7 @@
       <activation>
         <property>
             <!--Below formatting for dev-support/generate-hadoopX-poms.sh-->
-            <!--h2--><name>!hadoop.profile</name>
+            <!--h2--><name>2.0</name>
         </property>
       </activation>
       <dependencies>
diff --git a/hbase-rest/pom.xml b/hbase-rest/pom.xml
index 36c1b8c..b75a7e9 100644
--- a/hbase-rest/pom.xml
+++ b/hbase-rest/pom.xml
@@ -317,7 +317,6 @@
     <dependency>
       <groupId>org.codehaus.jackson</groupId>
       <artifactId>jackson-mapper-asl</artifactId>
-      <scope>test</scope>
     </dependency>
   </dependencies>
   <profiles>
@@ -385,6 +384,30 @@
       </build>
     </profile>
     <!-- Hadoop-specific dependencies -->
+    <!-- profile for building against CDH5. This is the default profile. -->
+    <profile>
+      <id>cdh5</id>
+      <activation>
+        <property>
+            <!--Below formatting for dev-support/generate-hadoopX-poms.sh-->
+            <!--h2--><name>!hadoop.profile</name>
+        </property>
+      </activation>
+      <dependencies>
+        <dependency>
+          <groupId>org.apache.hadoop</groupId>
+          <artifactId>hadoop-core</artifactId>
+        </dependency>
+        <dependency>
+          <groupId>org.apache.hadoop</groupId>
+          <artifactId>hadoop-common</artifactId>
+        </dependency>
+        <dependency>
+          <groupId>org.apache.hadoop</groupId>
+          <artifactId>hadoop-auth</artifactId>
+        </dependency>
+      </dependencies>
+    </profile>
     <!-- profile for building against Hadoop 2.0.x
          This is the default.
          -->
@@ -393,7 +416,7 @@
       <activation>
         <property>
           <!--Below formatting for dev-support/generate-hadoopX-poms.sh-->
-          <!--h2--><name>!hadoop.profile</name>
+          <!--h2--><name>2.0</name>
         </property>
       </activation>
       <dependencies>
diff --git a/hbase-server/pom.xml b/hbase-server/pom.xml
index f893cf3..06319c8 100644
--- a/hbase-server/pom.xml
+++ b/hbase-server/pom.xml
@@ -554,7 +554,6 @@
     <dependency>
       <groupId>org.hamcrest</groupId>
       <artifactId>hamcrest-core</artifactId>
-      <scope>test</scope>
     </dependency>
     <dependency>
       <groupId>org.apache.hadoop</groupId>
@@ -798,15 +797,15 @@
       <dependencies>
         <dependency>
           <groupId>org.apache.hadoop</groupId>
-          <artifactId>hadoop-common</artifactId>
+          <artifactId>hadoop-core</artifactId>
         </dependency>
         <dependency>
           <groupId>org.apache.hadoop</groupId>
-          <artifactId>hadoop-auth</artifactId>
+          <artifactId>hadoop-common</artifactId>
         </dependency>
         <dependency>
           <groupId>org.apache.hadoop</groupId>
-          <artifactId>hadoop-core</artifactId>
+          <artifactId>hadoop-auth</artifactId>
         </dependency>
         <dependency>
           <groupId>org.apache.hadoop</groupId>
@@ -826,6 +825,11 @@
           <artifactId>hadoop-minicluster</artifactId>
           <scope>test</scope>
         </dependency>
+        <dependency>
+          <groupId>org.apache.hadoop</groupId>
+          <artifactId>hadoop-test</artifactId>
+          <scope>test</scope>
+        </dependency>
       </dependencies>
       <build>
         <plugins>
diff --git a/pom.xml b/pom.xml
index 8ec12e1..9f4fcbb 100644
--- a/pom.xml
+++ b/pom.xml
@@ -1782,7 +1782,6 @@
         <groupId>org.hamcrest</groupId>
         <artifactId>hamcrest-core</artifactId>
         <version>${hamcrest.version}</version>
-        <scope>test</scope>
       </dependency>
       <dependency>
         <groupId>org.mockito</groupId>
-- 
1.7.9.5

